pkg/reconciler/configmappropagation/resources/configmap.go

```diff
@@ -46,12 +46,14 @@ func MakeConfigMap(args ConfigMapArgs) *corev1.ConfigMap {
 	}
 }
 
+// MakeCopyConfigMapName returns a unique name for a copy ConfigMap owned by the
+// ConfigMapPropagation with the given name.
 func MakeCopyConfigMapName(configMapPropagationName, configMapName string) string {
-	return configMapPropagationName + "-" + configMapName
+	return kmeta.ChildName(configMapPropagationName, "-"+configMapName)
 }
 
-// MakeCopyCOnfigMapLabel uses '-' to separate namespace and configmap name instead of '/',
-// for label values only accept '-', '.', '_'.
+// MakeCopyConfigMapLabel returns a unique label value for a copy ConfigMap to
+// reference its original ConfigMap.
 func MakeCopyConfigMapLabel(configMapPropagationNamespace, configMapName string) string {
-	return configMapPropagationNamespace + "-" + configMapName
+	return kmeta.ChildName(configMapPropagationNamespace, "-"+configMapName)
```

Comment:
The value of this label seems like it might be wrong, but I may be remembering incorrectly how it works. Naively I'd expect it to use the original CM's namespace instead of the CMP's namespace. @grac3gao can you shed some light on this?

Reply:
Yes, it should be the original namespace. It seems it was mistakenly named as current namespace + original configmap name when creating configmap resources. Would you mind changing the resources' corresponding label? Also, in reconciler logic, when iterate copied configmap, we get the corresponding original configmap name by `originalConfigMapName := strings.TrimPrefix(copyConfigMap.Name, cmp.Name+"-")`, it may not work by using childname for copied configmap